ext/windows/service/daemon.rb

```diff
@@ -127,23 +128,16 @@ def log(msg, level)
         File.open(LOG_FILE, 'a:UTF-8') { |f| f.puts("#{Time.now} Puppet (#{level}): #{msg}") }
       end
 
-      case level
-        when :debug, :info, :notice
-          report_windows_event(EVENTLOG_INFORMATION_TYPE,0x01,msg.to_s)
-        when :err, :alert, :emerg, :crit
-          report_windows_event(EVENTLOG_ERROR_TYPE,0x03,msg.to_s)
-        else
-          report_windows_event(EVENTLOG_WARNING_TYPE,0x02,msg.to_s)
-      end
+      native_type, native_id = Puppet::Util::Windows::EventLog.to_native(level)
+      report_windows_event(native_type, native_id, msg.to_s)
     end
   end
 
   def report_windows_event(type,id,message)
     begin
       eventlog = nil
-      eventlog = Win32::EventLog.open("Application")
+      eventlog = Puppet::Util::Windows::EventLog.open("Puppet")
```

Comment:
Yeah that one mystified me for a bit as I traced through win32/eventlog and the MSDN function reference, which is murky at best on this topic. Turns out, we've been a bit suboptimal I think. What `Win32::EventLog` does when we call `Win32::EventLog.open("Application")` is open a handle via [`OpenEventLog`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363672(v=vs.85).aspx) with an event source parameter of "Application" https://github.com/djberg96/win32-eventlog/blob/ffi2/lib/win32/eventlog.rb#L94-L128. According to the MSDN docs, this "Opens a handle to the specified event log." and in the notes:
```
lpSourceName [in]
  The name of the log.
    If you specify a custom log and it cannot be found, the event logging service opens the Application log; however, there will be no associated message or category string file.
```

Seems pretty reasonable so far. However, later, when we actually write the log entry via `Win32::EventLog#report_event`, what that method effectively does is open a _new_ event log handle with the newly supplied source parameter value (in our case, "Puppet"). It does so via [`RegisterEventSource`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363678(v=vs.85).aspx) (https://github.com/djberg96/win32-eventlog/blob/ffi2/lib/win32/eventlog.rb#L668). The previous handle that was opened by `Win32::EventLog::open` is never used by `Win32::EventLog#report_event`. Wha? Well according to the docs, `RegisterEventSource` "Retrieves a registered handle to the specified event log." In order to be a registered handle, the event source needs to have been previously added to the registry (which I believe we actually do with our MSI). Side note, this:

```
Remarks

If the source name cannot be found, the event logging service uses the Application log. Although events will be reported , the events will not include descriptions because there are no message and category message files for looking up descriptions related to the event identifiers.
```
That fallback is what was previously happening when we ran via gem or source (and is still the case after my PR).

The example C++ code here: https://msdn.microsoft.com/en-us/library/windows/desktop/aa363680(v=vs.85).aspx only calls `RegisterEventSource` to get the handle, and then report events through it - we don't need call `OpenEventSource`. In my other searches online this seemed to be consistent.

In summary, our use of the Win32 code was obtaining an extra initial event log handle (the "Application" handle) and then never doing anything with it. So my PR just skips that step entirely and gets the initial handle via `RegisterEventSource` with the source name, "Puppet", and reports through that handle.